src/EditorFeatures/CSharpTest/UseIsNullCheck/UseIsNullCheckForEqualityOperatorTests.cs

```diff
@@ -304,6 +304,16 @@ void M<T>(T s)
         if ([||](object)s == null)
             return;
     }
+}",
+@"using System;
+
+class C
+{
+    void M<T>(T s)
+    {
+        if ((object)s is null)
```

Comment:
this does not appear to be the desirable final form.  there shoul dbe no `(object)` cast.

Reply:
@dotnet/roslyn-compiler What version of C# is the first to support `is null` for an unconstrained generic type?

Reply:
Going by #34678 and #34695, LangVersion 8 is required